### ISSUE TITLE
UC mode detected in Pixelscan/CF

### ISSUE THREAD

**Issue by d3mon187** [seleniumbase/SeleniumBase#2407]
I had this working last week, so not sure if something changed or I broke it.  
I can pass Cloudflare after clicking the challenge, but I'm pretty sure I was able to get passed it without having to click before.   
Pixelscan will flag it as an "anti-detect" browser with inconsistent fingerprint.  If I run it using pytest and sbvirtualdisplay, then it will pass the browser check, but still flags as inconsistent fingerprint.

Running on ubuntu server with no gui.
Fails browser and fingerprint.
```
from seleniumbase import Driver

driver = Driver(uc=True)
driver.uc_open_with_reconnect("https://pixelscan.net", 10)
driver.save_screenshot('/opt/storage/chrome/testbss.png');
driver.quit()
```

**Comment by mdmintz** [comment]
Use `SB()` instead of `Driver()` so that you get the built-in virtual display that you need when running on a Linux Server that has no GUI:

```python
from seleniumbase import SB

with SB(uc=True, xvfb=True) as sb:
    sb.driver.uc_open_with_reconnect("https://pixelscan.net", 20)
```

<img width="1400" alt="Screenshot 2024-01-02 at 6 14 38 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/0f652a27-2ef1-489d-aec0-542e7edaf69e">


(Otherwise you'll need to use https://github.com/mdmintz/sbVirtualDisplay)

If for some reason `xvfb=True` didn't work for you, use `headless2=True` instead. From all the Linux environments I've seen, either `xvfb=True` OR `headless2=True` does the trick when using the `SB()` format.

If you're sticking with the `Driver()` format, then know how to use `sbvirtualdisplay` correctly:

```python
from sbvirtualdisplay import Display
from seleniumbase import Driver

display = Display(visible=0, size=(1440, 1880))
display.start()

driver = Driver(uc=True, headless=False, incognito=True)
try:
    driver.uc_open_with_reconnect("https://pixelscan.net", 20)
finally:
    driver.quit()

display.stop()
```

(That overrides the default headless mode on Linux and forces your script to go through the virtual display instead.)

**Comment by d3mon187** [comment]
I've tried using SB(), and your code above, and it's still flagging me.  Like I said, I could have sworn I was passing last week.  Also, oddly enough, I tried downgrading Chrome to 114 and that allowed me to pass the browser test when using sbvirtualdisplay but I still get bad fingerprint.  On Chrome 120, I always get badfingerprint and only pass browser using vxfb without virtualdisplay.

Here are my results running a simple "python script.py" found below:
no virtual + vxfb = True :  Good Browser, bad Fingerprint

sbvirtualdisplay + headless = false :  Bad Browser and Fingerprint
sbvirtualdisplay + xvfb = True :  Bad Browser and Fingerprint
no virtual + headless = True :  Bad Browser and Fingerprint
no virtual + headless2 = True :  Bad Browser and Fingerprint

Environment:
'Python': '3.8.10', 'Platform': 'Linux-5.4.0-169-generic-x86_64-with-glibc2.29', 'Packages': {'pytest': '7.4.4', 'pluggy': '1.3.0'}, 'Plugins': {'rerunfailures': '13.0', 'html': '2.0.1', 'metadata': '3.0.0', 'xdist': '3.5.0', 'ordering': '0.6', 'seleniumbase': '4.22.4'}
Chrome 120

Test code:
```
from sbvirtualdisplay import Display
from seleniumbase import SB

display = Display(visible=0, size=(1440, 1880))
display.start()

with SB(uc=True, headless=False, incognito=True) as sb:
    try:
        sb.driver.uc_open_with_reconnect("https://pixelscan.net", 20)
    finally:
        sb.sleep(10)
        sb.driver.save_screenshot('test-ss.png');
        sb.driver.quit()
display.stop()
```

**Comment by mdmintz** [comment]
Calling `sbvirtualdisplay` manually  is only for the `Driver()` manager.
`SB()` already includes `sbvirtualdisplay` automatically when `xvfb=True`.
Calling both `SB()` and `sbvirtualdisplay` together would likely cause issues.

Use only one of the below ways:

```python
from seleniumbase import SB

with SB(uc=True, xvfb=True) as sb:
    sb.driver.uc_open_with_reconnect("https://pixelscan.net", 20)
```

OR

```python
from sbvirtualdisplay import Display
from seleniumbase import Driver

display = Display(visible=0, size=(1440, 1880))
display.start()

driver = Driver(uc=True, headless=False, incognito=True)
try:
    driver.uc_open_with_reconnect("https://pixelscan.net", 20)
finally:
    driver.quit()

display.stop()
```

**Comment by d3mon187** [comment]
With those methods I still fail on the fingerprint being inconsistent, but do pass on valid browser.  I appreciate the help, but there must be something else missing with my setup or otherwise changed with pixelscan and CF.  Do I need any ini or config files in the rootdir of the script for it to be successful?

That does explain why the "sbvirtualdisplay + xvfb = True" test failed though, which is good to know.

**Comment by mdmintz** [comment]
As I mentioned before, this script works for me:

```python
from seleniumbase import SB

with SB(uc=True, xvfb=True) as sb:
    sb.driver.uc_open_with_reconnect("https://pixelscan.net", 20)
```

<img width="1400" alt="Screenshot 2024-01-02 at 6 14 38 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/0f652a27-2ef1-489d-aec0-542e7edaf69e">

**Comment by d3mon187** [comment]
Sorry to bother, but I've tried from scratch now on two more servers with the same result of bad fingerprint.  Could it be something to do with non-GUI Linux Server vs MacOS?

Here is my workflow:
ubuntu 20.04LTS
pip 23.3.2
python 3.8.10
apt install google-chrome-stable from "deb http://dl.google.com/linux/chrome/deb/ stable main"
current chrome 120.0.6099.199-1
apt install xvfb
cd to testfolder
python3 -m venv sbase_env
source sbase_env/bin/activate
pip install seleniumbase
python test_script.py
chromedriver 120.0.6099.109 ready

```
from seleniumbase import SB

with SB(uc=True, xvfb=True) as sb:
    try:
        sb.driver.uc_open_with_reconnect("https://nowsecure.nl", 30)
    finally:
        sb.sleep(60)
        sb.driver.save_screenshot('test-ss.png');
        sb.driver.quit()
```

I've tried this all a hundred different ways, different versions, sudo or non, virtual environment or not.  Read through seleniumbase.io multiple times.
Also, running from Windows server with gui on normal chrome indeed works fine from the same IP.

**Comment by mdmintz** [comment]
Then likely the version of Chrome you have gives away the OS info. If that OS specifically has no GUI, then that site could suspect something, even if they can't detect any automation usage directly.

**Comment by d3mon187** [comment]
Enjoy your vacay!  This is more just for documenting what I've found.

Looks like all headless methods are getting detected and only a headed session works to pass pixelscan.  I just tried running on a Windows Server with GUI and both headless and headless2 were flagged as bad browser.  Headed was successful on pixelscan, but still asks to click "verify you are human" on nowsecure.nl.

For years I've been able to pass with my own methods, but this past year I cannot keep up.  Thank you for your dedication!  We all appreciate it!
